packages/codegen-ui-react/lib/utils/forms/value-mappers.ts

```diff
@@ -0,0 +1,166 @@
+/*
+  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+  Licensed under the Apache License, Version 2.0 (the "License").
+  You may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+
+import { factory, NodeFlags, SyntaxKind } from 'typescript';
+
+/*
+given a timestamp number it will convert it to datetime local for forms
+first does a check to see if the timestamp is in seconds or milliseconds
+*/
+export const getDateTimeLocal = (ts: number) => {
+  if (Math.abs(Date.now() - ts) < Math.abs(Date.now() - ts * 1000)) {
+    return new Date(ts).toISOString().replace('Z', '');
+  }
+  return new Date(ts * 1000).toISOString().replace('Z', '');
```

Comment:
this isn't a "local" conversion so much as it's just stripping off the UTC marker. Why was this necessary, and are there any data integrity concerns here?

Reply:
This change is added so the date timestamp values being passed in are viewable in an update form.
This shouldn't change the date value itself as its only being used in defaultValue.
Not opposed to not supporting the defaultValue is there a concern where we don't want this value mapper?

Reply:
synced offline - reformatting the date so the local time is preserved only when presenting the value